src/main/host/syscall/handler/clone.rs

```diff
@@ -243,7 +237,25 @@ impl SyscallHandler {
         ctid: ForeignPtr<libc::pid_t>,
         newtls: u64,
     ) -> Result<libc::pid_t, SyscallError> {
-        Self::clone_internal(ctx, flags_and_exit_signal, child_stack, ptid, ctid, newtls)
+        let raw_flags = flags_and_exit_signal as u32 & !0xff;
+        let raw_exit_signal = (flags_and_exit_signal as u32 & 0xff) as i32;
+
+        let Some(flags) = CloneFlags::from_bits(raw_flags as u64) else {
+            warn!("Couldn't parse clone flags: {raw_flags:x}");
+            return Err(Errno::EINVAL.into());
+        };
```

Comment:
I think this should be "debug"? This isn't an issue of shadow not implementing something correctly, it's an issue of the caller not giving a valid flag.

Reply:
done